nixos/modules/config/qt5.nix

```diff
@@ -0,0 +1,108 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.qt5;
+
+  platformThemePackage = if cfg.platformTheme == "gnome"
+    then pkgs.qgnomeplatform
+    else pkgs.libsForQt5.qtstyleplugins;
+
+  stylePackage = if cfg.style == "adwaita"
+    then pkgs.adwaita-qt
+    else pkgs.libsForQt5.qtstyleplugins;
+
+in
+
+{
+
+  options = {
+    qt5 = {
+
+      enable = mkEnableOption "Qt5 themeing configuration";
+
+      platformTheme = mkOption {
+        type = types.nullOr (types.enum [
+          "gtk2"
+          "gnome"
+        ]);
+        default = null;
+        example = "gnome";
+        relatedPackages = [
+          "qgnomeplatform"
+          ["libsForQt5" "qtstyleplugins"]
+        ];
+        description = ''
+          Selects the platform theme to use for Qt5 applications.</para>
+          <para>The options are
+          <variablelist>
+            <varlistentry>
+              <term><literal>gtk</literal></term>
+              <listitem><para>Use GTK theme with
+                <link xlink:href="https://github.com/qt/qtstyleplugins">qtstyleplugins</link>
+              </para></listitem>
+            </varlistentry>
+            <varlistentry>
+              <term><literal>gnome</literal></term>
+              <listitem><para>Use GNOME theme with
+                <link xlink:href="https://github.com/FedoraQt/QGnomePlatform">qgnomeplatform</link>
+              </para></listitem>
+            </varlistentry>
+          </variablelist>
+        '';
+      };
+
+      style = mkOption {
+        type = types.nullOr (types.enum [
+          "adwaita"
+          "cleanlooks"
+          "gtk2"
+          "motif"
+          "plastique"
+        ]);
+        default = null;
+        example = "adwaita";
+        relatedPackages = [
+          "adwaita-qt"
+          ["libsForQt5" "qtstyleplugins"]
+        ];
+        description = ''
+          Selects the style to use for Qt5 applications.</para>
+          <para>The options are
+          <variablelist>
+            <varlistentry>
+              <term><literal>adwaita</literal></term>
+              <listitem><para>Use Adwaita Qt style with
+                <link xlink:href="https://github.com/FedoraQt/adwaita-qt">adwaita</link>
+              </para></listitem>
+            </varlistentry>
+            <varlistentry>
+              <term><literal>cleanlooks</literal></term>
+              <term><literal>gtk2</literal></term>
+              <term><literal>motif</literal></term>
+              <term><literal>plastique</literal></term>
+              <listitem><para>Use styles from
+                <link xlink:href="https://github.com/qt/qtstyleplugins">qtstyleplugins</link>
+              </para></listitem>
+            </varlistentry>
+          </variablelist>
+        '';
+      };
+    };
+  };
+
+  config = mkIf (cfg.enable == true && cfg.platformTheme != null && cfg.style != null) {
```

Comment:
Would it make sense to only use `mkIf cfg.enable { ... }` and remove the `nullOr` for `platformTheme` and `style`? This way if people do `enable = true;`, NixOS will automatically tell them they need to set those other two options. A default for those options might be appropriate too

Reply:
> Would it make sense to only use mkIf cfg.enable { ... } and remove the nullOr for platformTheme and style? This way if people do enable = true;, NixOS will automatically tell them they need to set those other two options.

Possibly, I'll see.

> A default for those options might be appropriate too

I'm pretty sure `platformTheme` and `style` in this module is completely preferential. A plasma user would be using `breeze` and that DE enforces it default.

Reply:
I see, maybe then a
```nix
{
  config = mkIf cfg.enable (mkMerge [
    (mkIf (cfg.platformTheme != null) {
      # ... cfg.platformTheme stuff
    })
    (mkIf (cfg.style != null) {
      # ... cfg.style stuff
    })
  ]);
}
```

Reply:
Well I mean they're preferential as in defaults don't make sense, It would be pretty pointless to set a theme engine and not a style that doesn't support it.  Though I'm seeing how this would silently do nothing, how about some asserts?

Reply:
Ah actually no defaults are needed if the options are just unset, so just removing the `nullOr` and the default, then users are forced to set something when they `enable = true;`. That sounds the most reasonable